runtime/codert_vm/cnathelp.cpp

```diff
@@ -1512,41 +1505,35 @@ old_slow_jitResolveFieldSetterDirect(J9VMThread *currentThread)
 static VMINLINE void*
 old_slow_jitResolveStaticFieldImpl(J9VMThread *currentThread, UDATA parmCount, J9ConstantPool *ramConstantPool, UDATA cpIndex, void *jitEIP, bool isSetter, bool direct)
 {
-	void *addr = NULL;
-	J9RAMStaticFieldRef *ramRef = (J9RAMStaticFieldRef*)ramConstantPool + cpIndex;
-	UDATA field = (UDATA)VM_VMHelpers::staticFieldAddressFromRef(ramRef, isSetter);
-	if (0 == field) {
-		J9JavaVM *vm = currentThread->javaVM;
-		J9Method *method = NULL;
-		UDATA resolveFlags = J9_RESOLVE_FLAG_RUNTIME_RESOLVE | J9_RESOLVE_FLAG_CHECK_CLINIT;
-		buildJITResolveFrameWithPC(currentThread, direct ? J9_SSF_JIT_RESOLVE : J9_SSF_JIT_RESOLVE_DATA, parmCount, true, 0, jitEIP);
-		if (isSetter) {
-			J9StackWalkState *walkState = currentThread->stackWalkState;
-			walkState->walkThread = currentThread;
-			walkState->skipCount = 0;
-			walkState->flags = J9_STACKWALK_VISIBLE_ONLY | J9_STACKWALK_COUNT_SPECIFIED;
-			walkState->maxFrames = 1;
-			vm->walkStackFrames(currentThread, walkState);
-			method = walkState->method;
-			resolveFlags |= J9_RESOLVE_FLAG_FIELD_SETTER;
-		}
-		field = (UDATA)vm->internalVMFunctions->resolveStaticFieldRef(currentThread, method, ramConstantPool, cpIndex, resolveFlags, NULL);
-		if ((UDATA)-1 == field) {
-			/* fetch result from floatTemp - stashed there in clinit case */
-			J9RAMStaticFieldRef *fakeRef = (J9RAMStaticFieldRef*)&currentThread->floatTemp1;
-			/* Resolve call indicated success - no need for extra putstatic checks, and the ref
-			 * must contain valid resolved information.
-			 */
-			field = (UDATA)VM_VMHelpers::staticFieldAddressFromRef(fakeRef, false);
-			Assert_CodertVM_false(0 == field);
```

Comment:
We lose the assert here - intentional?

Reply:
Yes, the helper used to return 0 for unresolved, now it only applies to resolved.